framework/src/mesh/PatternedMesh.C

```diff
@@ -0,0 +1,164 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "PatternedMesh.h"
+#include "Parser.h"
+#include "InputParameters.h"
+
+// libMesh includes
+#include "libmesh/mesh_modification.h"
+#include "libmesh/serial_mesh.h"
+#include "libmesh/exodusII_io.h"
+
+template<>
+InputParameters validParams<PatternedMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRequiredParam<std::vector<MeshFileName> >("files", "The name of the mesh files to read.  They are automatically assigned ids starting with zero.");
+
+  params.addParam<Real>("x_width", 0, "The tile width in the x direction");
+  params.addParam<Real>("y_width", 0, "The tile width in the y direction");
+  params.addParam<Real>("z_width", 0, "The tile width in the z direction");
+
+  // x boundary names
+  params.addParam<BoundaryName>("left_boundary", "left_boundary", "name of the left (x) boundary");
+  params.addParam<BoundaryName>("right_boundary", "right_boundary", "name of the right (x) boundary");
+
+  // y boundary names
+  params.addParam<BoundaryName>("top_boundary", "top_boundary", "name of the top (y) boundary");
+  params.addParam<BoundaryName>("bottom_boundary", "bottom_boundary", "name of the bottom (y) boundary");
+
+  // z boundary names
+  params.addParam<BoundaryName>("front_boundary", "front_boundary", "name of the front (z) boundary");
+  params.addParam<BoundaryName>("back_boundary", "back_boundary", "name of the back (z) boundary");
+
+  params.addRequiredParam<std::vector<std::vector<unsigned int> > >("pattern", "A double-indexed array starting with the upper-left corner");
+
```

Comment:
No class description? I came here because I do not understand what the meaning of "pattern" is in this context. It's this a mesh tiled form other meshes?